File: android/src/main/java/com/plugin/flutter/zsdk/PrinterErrorDetails.java
package com.plugin.flutter.zsdk;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by luis901101 on 2020-01-07.
 */
public class PrinterErrorDetails
{
    final StatusInfo statusInfo;
    final String message;

    public PrinterErrorDetails(StatusInfo statusInfo, String message)
    {
        this.statusInfo = statusInfo != null ? statusInfo : new StatusInfo(Status.UNKNOWN, Cause.UNKNOWN);
        this.message = message;
    }

    Map toMap() {
        Map map = new HashMap();
        map.put("statusInfo", statusInfo.toMap());
        map.put("message", message);
        return map;
    }
}

File: lib/src/printer_error_details.dart

import 'package:zsdk/src/cause.dart';
import 'package:zsdk/src/status.dart';
import 'package:zsdk/src/status_info.dart';

/// Created by luis901101 on 2020-01-07.
class PrinterErrorDetails
{
    StatusInfo statusInfo;
    String message;

    PrinterErrorDetails(this.statusInfo, this.message) {
        this.statusInfo ??= StatusInfo(Status.UNKNOWN, Cause.UNKNOWN);
        this.message = message;
    }

    factory PrinterErrorDetails.fromMap(Map<dynamic, dynamic> map) =>
        map != null ?
        PrinterErrorDetails(
            StatusInfo.fromMap(map['statusInfo']),
            map['message'],
        ) : null;
}
